cheat_client_detect.lua

```diff
@@ -0,0 +1,51 @@
+-- taken from https://github.com/S-S-X/dftest/blob/master/init.lua
+local has_beerchat = minetest.get_modpath("beerchat")
+local mp = minetest.get_modpath("beowulf")
+
+local client_hashes = {
+	"dragonfire"
+}
+
+-- Read hashes from generated files
+for _, filename in ipairs(minetest.get_dir_list(mp .. "/hashes")) do
+	local file = io.open(mp .. "/hashes/" .. filename)
+	for line in file:lines() do
+		client_hashes[#client_hashes + 1] = line:sub(line:find('^"%w+"'))
+	end
+	file:close()
+end
+
+minetest.log("action", "[beowulf] Loaded " .. #client_hashes .. " cheat client hashes")
+
+local function check_version_string(s)
+	for _, v in ipairs(client_hashes) do
+		if s:find(v) then
+			return v
+		end
+	end
+end
```

Comment:
For simple stats it would be nice if this function also returned source file name so it could be reported to logs and beerchat (if installed).

What do you think, should it be added right here on this PR? If so then likely simplest method would be to just make client_hashes nested with filename on root level and hashes under that, shouldn't really affect performance or anything and I think not too complex.

Reply:
I would prefer to have it in another PR, not a big deal though.